ggircs-app/app/components/FuelManagement/CarbonTaxActFuelType.tsx

```diff
@@ -0,0 +1,219 @@
+import { FuelSelectionComponent } from "./FuelSelectionComponent";
+import { FuelChargeRow } from "./FuelChargeRow";
+import CreateEditFuelChargeRow from "./CreateEditFuelChargeRow";
+import { useRouter } from "next/router";
+import Button from "@button-inc/bcgov-theme/Button";
+
+interface Props {
+  query: CarbonTaxActFuelType_query$key;
+  pageQuery: GraphQLTaggedNode;
+}
+
+const DEFAULT_RATE_PERIODS_TO_SHOW = 3;
+
+export const CarbonTaxActFuelType: React.FC<Props> = ({ query, pageQuery }) => {
+  const { carbonTaxActFuelType, allCarbonTaxActFuelTypes } = useFragment(
+    graphql`
+      fragment CarbonTaxActFuelType_query on Query
+      @argumentDefinitions(carbonTaxActFuelTypeId: { type: "ID!" }) {
+        carbonTaxActFuelType(id: $carbonTaxActFuelTypeId) {
+          id
+          rowId
+          carbonTaxFuelType
+          ctaRateUnits
+          fuelChargesByCarbonTaxActFuelTypeId(
+            first: 1000
+            orderBy: START_DATE_ASC
+          )
+            @connection(
+              key: "CarbonTaxActFuelType_fuelChargesByCarbonTaxActFuelTypeId"
+            ) {
+            __id
+            edges {
+              node {
+                id
+                startDate
+                endDate
+                ...FuelChargeRow_fuelCharge
+              }
+            }
+          }
+        }
+        allCarbonTaxActFuelTypes {
+          edges {
+            node {
+              id
+              carbonTaxFuelType
+            }
+          }
+        }
+      }
+    `,
+    query
+  );
+
+  const router = useRouter();
+
+  const isDateInFuelTypeFuelCharges = (date: string, fuelChargeId?: string) => {
+    const dateToValidate = new Date(date);
+    const invalidDate =
+      carbonTaxActFuelType.fuelChargesByCarbonTaxActFuelTypeId.edges
+        .filter((edge) => edge.node.id !== fuelChargeId)
+        .some(
+          (edge) =>
+            new Date(edge.node.startDate) <= dateToValidate &&
+            new Date(edge.node.endDate) >= dateToValidate
+        );
+    if (invalidDate) return false;
+
+    return true;
+  };
+
+  // Default the number of fuel charge rows shown to the last 3 rows in the list.
+  // Allow alternating between showing all rows or the last 3 rows with handleShowCharges().
+  const fromIndex =
+    router.query.showAll === "true"
+      ? 0
+      : carbonTaxActFuelType?.fuelChargesByCarbonTaxActFuelTypeId?.edges
+          ?.length - DEFAULT_RATE_PERIODS_TO_SHOW;
+  const showPeriodText =
+    router.query.showAll === "true"
+      ? "Show Less Rate Periods"
+      : `Show All Rate Periods (${carbonTaxActFuelType?.fuelChargesByCarbonTaxActFuelTypeId?.edges?.length})`;
+
+  const handleShowCharges = () => {
+    const url = {
+      pathname: router.pathname,
+      query: {
+        ...router.query,
+        showAll: router.query.showAll === "false" ? "true" : "false",
+      },
+    };
+    router.push(url, url, { shallow: true });
```

Comment:
I wonder if we should use `router.replace(...)` , no need to save these states in the browser history